### ISSUE TITLE
Listening port check is rather naive

### ISSUE THREAD

**Issue by the-maldridge** [qdm12/ddns-updater#335]
Some part of the code base is naively checking the effective UID and is dissallowing bind of system ports when not running as root.  This ignores the case where CAP_NET_BIND is available in the current context.

I can't find in a cursory search where the check happens in the code, otherwise I would have sent a patch.

**Comment by brzd** [comment]
Is this the reason that I am not able to change webUI port to other than 8000?

**Comment by qdm12** [comment]
@the-maldridge sure but what's your use case really? Why would you want to listen on privileged ports?

@brzd no, you can set the listening port between 1024 and 65535

**Comment by drewstopherlee** [comment]
Not sure if it's directly related, but I'm running into a similar issue when starting the container since updating to 2.5.0:
```
ERROR settings validation: resolver settings: splitting host and port from address: cannot use privileged ports (1 to 1023) when running without root: 53
```
Due to split DNS, I have to use the `RESOLVER_ADDRESS` environment variable and it kicks this error every time I restart the container. It's currently running on a Ubuntu Server 22.04 on a Raspberry Pi, so when I try to add `user: root` to the docker-compose, it fails to deploy. 
If this needs to be a separate issue, please let me know and I'll open one.

**Comment by drewstopherlee** [comment]
> @drewstopherlee unrelated-ish, but that's a bug that got introduced **after** v2.5.0 (on the latest image, you can check your logs for the version you're running). 
Yep, I noticed it and edited my comment right before you replied. Thanks so much!

**Comment by the-maldridge** [comment]
Been a while since I looked back into this since I worked around the bug via other means, but I'm refactoring some things and ran back into this. At its core that this is a bug due to incorrect assumptions around bind permissions.  A service need not be root nor even have any capabilities to bind "privileged" ports, and given that its entirely feasible to adjust the port boundary or remove it entirely, the check is flawed at best.

Since you wanted a use case though I can explain what I'm currently refactoring to, which is that I have containers using a dynamic macvlan architecture for networking where BGP is announcing a service VIP per-host per-container based on wherever the container happens to be at the moment.  This is the one container in my entire fleet which does not put a webserver on port 80, and it is the last one I have notes for that I have to access it on a non-standard port.  I guess I can kludge around it by running an nginx sidecar and having that map the traffic, but that's pretty clunky.

**Comment by qdm12** [comment]
This makes sense, thanks for sharing! I'll get to it right now, sorry for the delay.

**Comment by qdm12** [comment]
Interesting; I guess I can change the validation library to just assume it's 1024 if we get a permission denied. Have you tried `cat /proc/sys/net/ipv4/ip_unprivileged_port_start` on your host?

**Comment by xiaoxiyao** [comment]
>  Have you tried `cat /proc/sys/net/ipv4/ip_unprivileged_port_start` on your host?

Permission denied too

**Comment by qdm12** [comment]
Even as root, for example using `sudo`?

**Comment by qdm12** [comment]
Also what does this command give

```
ls -l /proc/sys/net/ipv4/ip_unprivileged_port_start
```

**Comment by xiaoxiyao** [comment]
I can't use sudo without rooting my phone，and `ls` also denied

**Comment by qdm12** [comment]
> I can't use sudo without rooting my phone

This makes sense; I've updated the settings library to handle "permission denied" as just assume the start is `1024`, can you please confirm it's working:

- if you're using a container, just pull the latest image `qmcgaw/ddns-updater` and restart the container
- if you're using the binary program, build the latest program using `go build -o ddns-updater github.com/qdm12/ddns-updater/cmd/ddns-updater`, and you can cross compile it by prefixing the `go build` command for example with `GOARCH=arm64 ` or `GOARCH=arm GOARM=7 `, depending on your phone CPU architecture.
